app/src/main/kotlin/com/saltedge/authenticator/tools/LogTools.kt

```diff
@@ -23,49 +23,39 @@ package com.saltedge.authenticator.tools
 import android.util.Log
 import com.google.firebase.crashlytics.FirebaseCrashlytics
 import com.saltedge.authenticator.BuildConfig
-
-private const val MAX_LOG_SIZE = 1000
-
-/**
- * Show log/logs with given message
- *
- * @receiver Throwable
- * @param message - the message that appears in the log
- */
-fun Throwable.log(message: String = "") {
-    try {
-        if (BuildConfig.DEBUG) {
-            printToLogcat("${this.cause}", message)
-            printStackTrace()
-        } else FirebaseCrashlytics.getInstance().recordException(this)
-    } catch (e: Exception) {
-        FirebaseCrashlytics.getInstance().recordException(e)
-    }
-}
+import timber.log.Timber
+import timber.log.Timber.DebugTree
 
 /**
  * Create an error collector for the application not for DEBUG version
  *
  * @return crashlytics
  */
 fun createCrashlyticsKit() {
-    FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(!BuildConfig.DEBUG)
+    if (BuildConfig.DEBUG) {
+        Timber.plant(DebugTree())
+    } else {
+        FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(!BuildConfig.DEBUG)
+        Timber.plant(CrashlyticsReportingTree())
+    }
 }
 
 /**
- * Show log/logs with given message and tag
+ * Show log/logs with given message
  *
- * @param tag - the tag that appears in the log
  * @param message - the message that appears in the log
  */
-fun printToLogcat(tag: String, message: String) {
-    if (BuildConfig.DEBUG && message.isNotEmpty()) {
-        var start = 0
-        while (start < message.length) {
-            var end = start + MAX_LOG_SIZE
-            if (end > message.length) end = message.length
-            Log.d(tag, message.substring(start, end).trim())
-            start += MAX_LOG_SIZE
+class CrashlyticsReportingTree : Timber.Tree() {
+
+    override fun log(priority: Int, tag: String?, message: String, throwable : Throwable?) {
```

Comment:
remove extra space after `throwable`.  
`message` is optional (see http://jakewharton.github.io/timber/timber/log/Timber.Tree.html#log-int-java.lang.String-java.lang.String-java.lang.Throwable-)

Reply:
2. https://github.com/JakeWharton/timber/commit/cf676130541ae9912b15fa66ee2eacee7220ce07 discussed in mattermost